website/docs/chapter-02/05_keyword_interface.md

```diff
@@ -0,0 +1,464 @@
+
+(*) Understanding keyword tags is not part of the syllabus.
+
+The following keywords are part of the Standard Libraries of Robot Framework.
+Their documentation has been generated by the Robot Framework tool `libdoc` which is included in Robot Framework.
+
+### 2.5.1.1 Example Keyword `Should Be Equal`
+
+[Documentation of `Should Be Equal` from `BuiltIn` library](https://robotframework.org/robotframework/latest/libraries/BuiltIn.html#Should%20Be%20Equal)
+
+`Should Be Equal` is part of the BuiltIn library and is documented as follows:
+
+![Should Be Equal Keyword Documentation](/img/Should_Be_Equal_Docs.png)
+
+This keyword has 2 "Mandatory Arguments" and 6 "Optional Arguments".
+All of them can be called positionally or by name.
+
+
+### 2.5.1.2 Example Keyword `Run Process`
+
+[Documentation of `Run Process` from `Process` library](https://robotframework.org/robotframework/latest/libraries/Process.html#Run%20Process)
+
+`Run Process` is part of the Process library and is documented as follows:
+
+![Run Process Keyword Documentation](/img/Run_Process_Docs.png)
+
+This keyword has one "Mandatory Arguments" `command` which can be called positionally or by name.
+The latter two arguments are optional.
+
+The argument `arguments` is a "Variable Number of Positional Arguments" and can only be set by position.
+Therefore, if it shall be set, all preceding arguments must be set by position as well.
+See [2.5.2.5 Variable Number of Positional Arguments](chapter-02/05_keyword_interface.md#2525-variable-number-of-positional-arguments) for more information about this kind of argument.
+
+The argument `configuration` is a "Free Named Argument" and can only be set by names.
+See [2.5.2.7 Free Named Arguments](chapter-02/05_keyword_interface.md#2527-free-named-arguments) for more information about this kind of argument.
+
+
+### 2.5.1.3 Example Keyword `Get Regexp Matches`
+
+[Documentation of `Get Regexp Matches` from `String` library](https://robotframework.org/robotframework/latest/libraries/String.html#Get%20Regexp%20Matches)
+
+`Get Regexp Matches` is part of the String library and is documented as follows:
+
+![Get Regexp Matches Keyword Documentation](/img/Get_Regexp_Matches_Docs.png)
+
+This keyword has 2 "Mandatory Arguments" that can be called positionally or by name.
+The last two arguments are optional.
+
+The argument `groups` is a "Variable Number of Positional Arguments" and can only be set by position.
+Therefore, if it shall be set, all preceding arguments must be set by position as well.
+See [2.5.2.5 Variable Number of Positional Arguments](chapter-02/05_keyword_interface.md#2525-variable-number-of-positional-arguments) for more information about this kind of argument.
+
+The argument `flags` is a "Named-Only Argument" and can only be set by name.
+See [2.5.2.6 Named-Only Arguments](chapter-02/05_keyword_interface.md#2526-named-only-arguments) for more information about this kind of argument.
+
+
+## 2.5.2 Keyword Arguments
```

Comment:
Each subsection for different argument types would benefit from one or more examples. Probably example of libdoc keyword documentation would be enough. Currently, this is probably one of the most abstract sections of the syllabus. In theory. In practice one does not really need to think about these (outside designing and implementing keywords, of course).